src/pudl/transform/epacems.py

```diff
@@ -209,37 +210,37 @@ def correct_gross_load_mw(df):
     return df
 
 
-def transform(epacems_raw_dfs, pudl_engine):
+@op(
+    required_resource_keys={"pudl_engine"}
+)
+def transform(context, raw_df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
The required resources are defined explicitly, but the required inputs here are defined implicitly? Are they entirely defined by the construction of the `@job` in `dagster_etl.py`? Do they only exist in the context of a `graph` or `job` definition?

Reply:
I think you can use resources where ever you want as long as you specify they in the @job decorator.